ikarus/utils/traversal.hh

```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+namespace Ikarus::utils {
+
+/**
+ * \brief A function which loops over all the nodes of a tree and performs different actions
+ * for a power node and a leaf node depending on the corresponding functor passed.
```

Comment:
Can you also indicate that this is not for all power nodes but only for those with leafs as childs.